src/OrchardCore/OrchardCore.Infrastructure/Shells.Database/Extensions/DatabaseShellsOrchardCoreBuilderExtensions.cs

```diff
@@ -7,14 +8,23 @@ namespace Microsoft.Extensions.DependencyInjection
     public static class DatabaseShellsOrchardCoreBuilderExtensions
     {
         /// <summary>
-        /// Host services to load shells settings and configuration from database.
+        /// Registers host services to load shells settings and configuration from the database,
+        /// but only if the related options are provided by the application level configuration.
         /// </summary>
         public static OrchardCoreBuilder AddDatabaseShellsConfiguration(this OrchardCoreBuilder builder)
         {
-            var services = builder.ApplicationServices;
+            var options = builder.Configuration
```

Comment:
What concerns me a little here @jtkech is that it will just silently fail if the configuration is set incorrectly. 

And then you would think you are using it, only to discover that you aren't